coercion.ts

```diff
@@ -51,81 +53,61 @@ export function coerceFile(file: unknown) {
   return file;
 }
 
-type WrapOption = {
-  wrap?: WrapSchema;
-  cache?: Map<AllSchema, AllSchema>;
-};
 /**
  * Reconstruct the provided schema with additional preprocessing steps
  * This coerce empty values to undefined and transform strings to the correct type
  */
-export function enableTypeCoercion<Type extends AllSchema,>(
-  type: Type,
-  options?: WrapOption,
-): AllSchema {
-  const cache = options?.cache ?? new Map<Type, AllSchema>();
-  const result = cache.get(type);
-
-  // Return the cached schema if it's already processed
-  // This is to prevent infinite recursion caused by z.lazy()
-  if (result) {
-    return result;
-  }
-
-  // A schema that does not have a type property does not exist.
-  // However, in the wrapped property such as optional schema,
-  // it is necessary to receive the BaseSchema expression,
-  // so in order to receive it, it is also possible to receive the type in BaseSchema.
-  if (!("type" in type)) {
-    return type;
-  }
-
-  let schema: AllSchema = type;
+export function enableTypeCoercion<T extends AllSchema>(
+  type:
+    | T
+    | SchemaWithPipe<[T, ...PipeItem<unknown, unknown, BaseIssue<unknown>>[]]>,
```

Comment:
I modified the type to accept a schema with a pipe function, but since it is no longer possible to determine the schema passed in, I set `@ts-expect-error`.